sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/ViewportPlugin.kt

```diff
@@ -0,0 +1,157 @@
+import com.mapbox.maps.plugin.viewport.experimental.state.OverviewViewportState
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.transition.DefaultViewportTransition
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransition
+
+/**
+ * The Viewport plugin allows to track objects on a map.
+ *
+ * It provides a structured approach to organizing camera management logic into states and transitions between them.
+ *
+ * at any given time, the viewport is either:
+ *  - idle (not updating the camera)
+ *  - in a state (camera is being managed by a ViewportState)
+ *  - transitioning (camera is being managed by a ViewportTransition)
+ */
+interface ViewportPlugin : MapPlugin {
+
+  /**
+   * Returns current [ViewportStatus].
+   *
+   * If current status is IDLE, returns ViewportStatus.State(null).
+   * @see addStatusObserver
+   */
+  val status: ViewportStatus
+
+  /**
+   * Executes a transition to requested state.
+   *
+   * When started, goes to [ViewportTransition]
+   * and to the final [ViewportState] when ended.
+   *
+   * If transition is canceled, state goes to IDLE.
+   *
+   * @param targetState The target [ViewportState] to transition to.
+   * @param completionListener The listener to observe the completion state.
+   */
+  fun transitionTo(targetState: ViewportState, completionListener: CompletionListener? = null)
+
+  /**
+   * Immediately goes to IDLE state canceling all ongoing transitions.
+   */
+  fun toIdle()
+
+  /**
+   * Indicates whether to transition [ViewportState] to IDLE (null) when user interact with the map
+   * using gestures.
+   *
+   * Defaults to true.
+   */
+  var transitionsToIdleUponUserInteraction: Boolean
+
+  // States
+
+  /**
+   * Returns list of registered states.
+   */
+  val states: Set<ViewportState>
+
+  /**
+   * Add a viewport state to the viewport plugin, which could be reused later with the state id.
+   *
+   * @param state the view port state to be added.
+   */
+  fun addState(state: ViewportState)
+
+  /**
+   * Remove a viewport state from the viewport plugin, and it could not be reused later with the state id.
+
+   * @param state the view port state to be removed.
+   */
+  fun removeState(state: ViewportState)
+
+  // Transitions
+
+  /**
+   * DefaultViewportTransition with default options.
+   *
+   * This transition is used unless overridden by one of the registered transitions.
+   */
+  var defaultTransition: ViewportTransition
+
+  /**
+   * Set the [ViewportTransition] for the transition from given [ViewportState] to target [ViewportState]
+   *
+   * @param transition The transition to be set.
+   * @param from The state before the transition.
+   * @param to The state after the transition.
+   */
+  fun setTransition(transition: ViewportTransition, from: ViewportState?, to: ViewportState)
+
+  /**
+   * Get the transition from the previous state and target state.
```

Comment:
```suggestion
   * Get the transition from given state and target state.
```

to correspond to `setTransition` above